contracts/carbon/interfaces/ICarbonController.sol

```diff
@@ -20,6 +20,11 @@ interface ICarbonController is IUpgradeable {
      */
     function tradingFeePPM() external view returns (uint32);
 
+    /**
+     * @dev returns the overriden trading fee for a given pair (in units of PPM)
+     */
+    function tradingFeePPMOverrides(uint128 pairId) external view returns (uint32);
```

Comment:
I think that the main problem here (and in several other places) is that you're exposing what used to be an internal implementation detail up until now - `pairId` - as a new part in the API.

Consider keep using a pair of tokens as the formal identifier (in all interface functions, events, etc).

Reply:
Pair ID is an identifier for the token pair, correct ?
Each token pair is associated with two token addresses. In a mapping I can't use two tokens, so it makes sense to have the mapping use pair id as a key and return the fee for that pair. 
It makes sense to keep the interface consistent with the view function and the event, since the mapping uses the pair id.

Keep in mind if that is changed, it will cost a SLOAD for the pair for both the view function and the event (Have to look up the tokens by pair id first).

Reply:
`Pair ID is an identifier for the token pair, correct ?` - an internal implementation detail.
Externally, the pair itself is used as the identifier.
I did not mean to complicate the implementation, only to pass the pair of tokens as part of the interface.
Inside the function, you can easily extract the pair ID.

Reply:
I would prefer to keep this interface function for consistency reasons.
What I can do is add an override to this function which accepts the token pair and returns the custom fee.
Open for discussion.

Reply:
I get what you mean, one step will be skipped (querying `pair(token0, token1)` in order to get the pair id when calling the first two functions).
For the event, it definitely makes sense to change since there is no public `getPairByPairId` function - so the event won't really help in tracing which pair has been updated unless we use an indexing service to track and store the pairs created.

I think this is worth implementing.

Reply:
Replaced pair id with a token pair for each the two functions and event.